appengine/genetics/js/visualization.js

```diff
@@ -324,187 +535,218 @@ Genetics.Visualization.processCageEvents_ = function() {
   var getMouseName = Genetics.Visualization.getMouseName;
   while (Genetics.Cage.EVENTS.length) {
     var event = Genetics.Cage.EVENTS.shift();
+
+    var mouse = (event['ID'] != null) ?
+        Genetics.Visualization.mice_[event['ID']] : null;
```

Comment:
event['ID'] can be 0 which would cause mouse to be set to 0 instead of getting the mouse. I car rename the properties to be more descriptive (MOUSE_ID, OPT_OPPONENT_ID)